README.md

```diff
@@ -1,24 +1,249 @@
-[![Coverage Status](https://coveralls.io/repos/apiaryio/gavel.js/badge.svg?branch=master)](https://coveralls.io/r/apiaryio/gavel.js?branch=master)
-[![Known Vulnerabilities](https://snyk.io/test/npm/gavel/badge.svg)](https://snyk.io/test/npm/gavel)
+<br />
 
-![Gavel.js - Validator of HTTP Transactions](https://raw.github.com/apiaryio/gavel/master/img/gavel.png?v=1)
+<p align="center">
+  <img src="https://raw.githubusercontent.com/apiaryio/gavel/master/img/gavel.png?v=1" alt="Gavel logo" />
+</p>
 
-Gavel detects important differences between actual and expected HTTP transactions (HTTP request and response pairs). Gavel also decides whether the actual HTTP transaction is valid or not.
+<h1 align="center">Gavel</h1>
 
-## Installation
+<p align="center">Gavel tells you whether an actual HTTP message is valid against an expected HTTP message.</p>
 
-```sh
-$ npm install gavel
+## Install
+
+```bash
+npm install gavel
+```
+
+## Usage
+
+### CLI
+
+```bash
+# (Optional) Record HTTP messages
+curl -s --trace - http://httpbin.org/ip | curl-trace-parser > expected
+curl -s --trace - http://httpbin.org/ip | curl-trace-parser > actual
+
+# Perform the validation
+cat actual | gavel expected
+```
+
+> **Gavel CLI is not supported on Windows**. Example above uses [`curl-trace-parser`](https://github.com/apiaryio/curl-trace-parser).
+
+### NodeJS
+
+```js
+const gavel = require('gavel');
+
+// Define HTTP messages
+const expected = {
+  statusCode: 200,
+  headers: {
+    'Content-Type': 'application/json'
+  }
+};
+
+const actual = {
+  statusCode: 404,
+  headers: {
+    'Content-Type': 'application/json'
+  }
+};
+
+// Perform the validation
+const result = gavel.validate(expected, actual);
+```
+
+The code above would return the following validation `result`:
+
+```js
+{
+  valid: false,
+  fields: {
+    statusCode: {
+      valid: false,
+      kind: 'text',
+      values: {
+        expected: '200',
+        actual: '404'
+      },
+      errors: [
+        {
+          message: `Expected status code '200', but got '404'.`,
+          values: {
+            expected: '200',
+            actual: '404'
+          }
+        }
+      ]
+    },
+    headers: {
+      valid: true,
+      kind: 'json',
+      values: {
+        expected: {
+          'Content-Type': 'application/json'
+        },
+        actual: {
+          'Content-Type': 'application/json'
+        }
+      },
+      errors: []
+    }
+  }
+}
+```
+
+### Usage with JSON Schema
+
+> When a parsable JSON body is expected without an explicit schema the [default schema](https://github.com/apiaryio/gavel-spec/blob/master/features/expectations/bodyJsonExample.feature) is inferred.
+
+You can describe the body expectations using [JSON Schema](https://json-schema.org/) by providing a valid schema to the `bodySchema` property of the expected HTTP message:
+
+```js
+const gavel = require('gavel');
+
+const expected = {
+  bodySchema: {
```

Comment:
I've discovered Gavel supports `bodySchema` as an object literal, so for JavaScript it doesn't have to be a string. It does support string as well, so you can wrap it. I've left an easier to digest example in the README not to increase the usage complexity.

Since Gavel operates on JSON Schema, I wouldn't say stringification is required. Perhaps we can mention this if Gavel starts to support other schema formats. What do you think?

Reply:
OK so it doesn't matter now and if we ever force it to be string for type consistency, we can just change the recommended way in the example. But perhaps we never force and just support both for convenience.